authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/GarminSourceAuthorizationService.kt

```diff
@@ -18,25 +18,37 @@ package org.radarbase.authorizer.service
 
 import com.fasterxml.jackson.databind.ObjectMapper
 import okhttp3.OkHttpClient
+import org.glassfish.jersey.process.internal.RequestScope
 import org.radarbase.authorizer.RestSourceClients
 import org.radarbase.authorizer.api.RestOauth1AccessToken
 import org.radarbase.authorizer.api.RestOauth1UserId
-import org.radarbase.authorizer.api.RestOauth2AccessToken
+import org.radarbase.authorizer.service.DelegatedRestSourceAuthorizationService.Companion.GARMIN_AUTH
 import org.radarbase.authorizer.api.RestSourceUserMapper
 import org.radarbase.authorizer.doa.RestSourceUserRepository
 import org.radarbase.jersey.exception.HttpBadGatewayException
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
+import java.util.*
 import javax.ws.rs.core.Context
 
 class GarminSourceAuthorizationService(
     @Context private val restSourceClients: RestSourceClients,
     @Context private val httpClient: OkHttpClient,
     @Context private val objectMapper: ObjectMapper,
     @Context private val userRepository: RestSourceUserRepository,
-    @Context private val userMapper: RestSourceUserMapper
+    @Context private val userMapper: RestSourceUserMapper,
+    @Context private val requestScope: RequestScope
 ): OAuth1RestSourceAuthorizationService(restSourceClients, httpClient, objectMapper, userRepository, userMapper) {
     val GARMIN_USER_ID_ENDPOINT = "https://healthapi.garmin.com/wellness-api/rest/user/id"
+    val DEREGISTER_CHECK_PERIOD = 5000L
+
+    init {
+        // This schedules a task that periodically checks users with elapsed end dates and deregisters them.
+        val task = object : TimerTask() {
+            override fun run() = checkForUsersWithElapsedEndDateAndDeregister()
+        }
+        Timer().scheduleAtFixedRate(task, 0, DEREGISTER_CHECK_PERIOD)
```

Comment:
You can use the ScheduledExecutorService from Jersey to run this task and have proper thread cleanup. In the constructor, add
```kotlin
@Context @BackgroundScheduler private val scheduler: ScheduledExecutorService,
```
and use that to schedule the task
```kotlin
scheduler.scheduleAtFixedRate({
    checkForUsersWithElapsedEndDateAndDeregister()
}, 0, DEREGISTER_CHECK_PERIOD, TimeUnit.MILLISECONDS)
```

Checking this every 5 seconds seems a lot though, perhaps that could be increased or made configurable?